### ISSUE TITLE
some missing devhelp packages

### ISSUE THREAD

**Issue by MuhannadAlrusayni** [Valadoc/valadoc-org#61]
Hi, there is some missing devhelp packages in valadoc.org:
- [x] appstream
- [x] gio-2.0
- [x] glib-2.0
- [x] gxml-0.14
- [x] gvncpluse-1.0
- [x] gvnc-1.0
- [x] gobject-2.0
- [ ] libgdaui-5.0
- [ ] libgitg-ext-1.0

**Comment by MuhannadAlrusayni** [comment]
it seems these packages are missing from valadoc.org but not in the mirror(valadata.org).
note : libgdaui-5.0 are missed from valadoc.org and valadata.org.

**Comment by btkostner** [comment]
All of those packages (as of right this moment) minus `gio-2.0`, `glib-2.0`, and `gobject-2.0` have a problem when you build the docs. When running `make build-docs` you get this output:

```
btkostner@Caprica:~/Projects/Web/valadoc-org$ make build-docs
rm -f -r tmp/
./generator \
	--vapidir /usr/share/vala-0.34/vapi/ \
	--vapidir "extra-vapis/" --vapidir "girs/vala/vapi/" \
	--driver 0.34 \
	--prefix "stable" \
	--target-glib 2.99 \
	--download-images \
	--skip-existing \
	--no-check-certificate \
	--all
...
skip 'cpufreq' ...
create 'gvncpulse-1.0' ...
  load keyfile ...
  select .gir:            girs/gir-1.0/GVncPulse-1.0.gir
  run valadoc ...
ERROR: Can't generate documentation for gvncpulse-1.0. See LOG for details.
Exit status 256 != 0
...
create 'gxml-0.10' ...
  run valadoc ...
ERROR: Can't generate documentation for gxml-0.10. See LOG for details.
Exit status 256 != 0
...
create 'appstream' ...
  select .gir:            girs/gir-1.0/AppStream-1.0.gir
  download images (gir) ...
  run valadoc ...
ERROR: Can't generate documentation for appstream. See LOG for details.
Exit status 256 != 0
...
create 'gvnc-1.0' ...
  load keyfile ...
  select .gir:            girs/gir-1.0/GtkVnc-2.0.gir
  run valadoc ...
ERROR: Can't generate documentation for gvnc-1.0. See LOG for details.
Exit status 256 != 0
skip 'xcb-composite' ...
skip 'xcb-xinerama' ...
skip 'dconf' ...
skip 'gdl-1.0' ...
skip 'libqpid-proton' ...
create 'libgdaui-5.0' ...
  select .gir:            girs/gir-1.0/Gdaui-5.0.gir
  run valadoc ...
ERROR: Can't generate documentation for libgdaui-5.0. See LOG for details.
Exit status 256 != 0
...
generate index ...
```

I'll work on tracking down the issues and fixing them, or reporting them upstream if need be.

**Comment by MuhannadAlrusayni** [comment]
Hi, it seemes that valadata.org is not avalible and these devhelp package is still not avalible too (in valadoc.org).
this is not good for pepole doesn't allways have internet access.

and I don't really know how to generate these devhelp packages to help solve this problem.
so it would be very helpful if anyone could solve this problem, since these libararies are core libararies.

**Comment by astavale** [comment]
@MuhannadAlrusayni `valadata.org` never existed, you may be thinking of `valadoc.valadate.org`. This was a temporary mirror of `valadoc.org`. It looks like the person hosting that mirror has taken it down now that `valadoc.org` is working fully.

If it's any help Linux distributions will carry the devhelp packages for the C API. For example with Fedora you can install `glib2-doc`. This package supplies devhelp files for gio-2.0, glib-2.0 and gobject-2.0. These are for the C API, the same as online at `developer.gnome.org`.

**Comment by MuhannadAlrusayni** [comment]
@astavale Oh, I thought it was valadata.org.

I'm doing it this way right now, when I have no internet access I go and look at C API in devhelp and try to figure out how to write it with vala, but that take's more time, and imagen if you work in project and that project needs a long time working on it, so looking to C API will not be helpful and will take even longer.
And this will not be easy for programmer who doesn't know C.

**Comment by btkostner** [comment]
This will probably take a while to fix. Attaching this issue for right now: https://bugzilla.gnome.org/show_bug.cgi?id=778226

**Comment by astavale** [comment]
@MuhannadAlrusayni There was a second mirror, beside valadoc.valadate.org, that is still up: http://valadoc.roojs.com/
From a quick look I can see devhelp for:
gio-2.0
glib-2.0
gobject-2.0

**Comment by MuhannadAlrusayni** [comment]
@astavale Great to have these packages :D, but it's weird this mirror is not listed somewhere in Vala docs and it seemes it's not up-to-date mirror for example the gobject page is a bit different not sure if it's in ordering things in the page or there are some different.

valadoc.org/gobject-2.0/index.htm
valadoc.roojs.com/#!wiki=gobject-2.0/index

anyway it's good temperary mirror until fix this bug in valadoc.org.
@astavale thanks for the mirror, by the way where did you find it ?

**Comment by btkostner** [comment]
@MuhannadAlrusayni I wasn't sure if they were going to close the issue and point me to upstream, so I only filed one issue right now. 

Also, that mirror is not up to date with _this_ master, but it does have some improvements, most of which have been put back into this repo.

**Comment by btkostner** [comment]
I just applied the above patch, and rebuilt the site. The mentioned packages that are check marked are currently fixed on valadoc.org.

**Comment by astavale** [comment]
That's great. This also resolves issue #59 AppStream present in menu but unavailable.
gxml-0.10 is already covered by #57 Documentation for gxml-0.10 not present.
Unfortunately gio-2.0, glib-2.0 and gobject-2.0 still do not have devhelp packages.

**Comment by MuhannadAlrusayni** [comment]
Hi, libgdaui-5.0 stil not avalible and libgitg-ext-1.0 added to the list.

**Comment by albfan** [comment]
gitg maintainer here, what should I fix to get libgitg-ext doc on valadoc.org?

**Comment by benwaffle** [comment]
@albfan You should edit packages.xml.

Here is an example: https://github.com/Valadoc/valadoc-org/pull/187/files

The vapi & gir should be part of vala, https://github.com/nemequ/vala-girs, or https://github.com/nemequ/vala-extra-vapis

Also, see the README https://github.com/Valadoc/valadoc-org#add-new-packages

**Comment by rkraneis** [comment]
libgee-0.8 devhelp leads to 404 https://valadoc.org/gee-0.8/gee-0.8.tar.bz2

**Comment by astavale** [comment]
@rkraneis this is something to do with #182 From what I can tell all links to a devhelp package return 404. This started after the last deploy and needs to be fixed before any work can start on getting libgitg-ext to work. Strangely following the link directly from your comment does download the devhelp file. Is some Javascript intercepting the links in Valadoc.org? The tooltips aren't working too well either.

@benwaffle libgitg-ext is already in packages.xml, https://valadoc.org/libgitg-ext-1.0/index.htm is there. It is the devhelp package that is the problem and it looks like the devhelp package itself is there as well: https://valadoc.org/libgitg-ext-1.0/libgitg-ext-1.0.tar.bz2 It is the link that is the problem, probably the same problem as #182

**Comment by benwaffle** [comment]
for some reason searching GitgExt shows nothing. out-of-date search index? @btkostner
